pkg/operator/ceph/csi/version.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// CephCSIVersion represents the Ceph CSI version format
+type CephCSIVersion struct {
+	Major  int
+	Minor  int
+	Bugfix int
+}
+
+var (
+	//supportedCSIVersions are versions that rook supports
+	supportedCSIVersions = []CephCSIVersion{{2, 0, 0}}
```

Comment:
Instead of an explicit list of supported versions I still think we need it to be a minimum csi version. If like this, as soon as CSI v2.1 is released, Rook will not support it by default. Rook should at least support newer minor versions by default. It would be good to follow the pattern that we're using for Ceph versions. Perhaps the Ceph version code can be re-used with the CSI version checking found [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/version/version.go).

Reply:
To summarize, what about this behavior?
- Set the minimum CSI version to v2.0. 
- Even if you set the flag `allowUnsupportedImage`, we still wouldn't allow anything less than the minimum version.
- All minor versions of supported major CSI versions are supported
- v2 would be the only supported major version of CSI currently

Reply:
What is the purpose of `allowUnsupportedImage` if we still do a version check?

Reply:
@BlaineEXE and I had a discussion about the approach. What about this?
- Rook should enforce a minimum version of the CSI driver. We know it just won't work with older versions of CSI. For now, setting a minimum version of v2.0 gives us a good baseline. Therefore, we need to check the version of the CSI driver every time we run to make sure it meets the min version
- Rook can only support the version where our integration tests have run. So I see the value in only supporting the current minor version, which is v2.0. When v2.1 is released, early adopters would need to use it as unsupported. Soon after that, we would update Rook to test against v2.1, add it to our supported list, and likely make it our default.

Reply:
OK. Got it.
The current approach only allows images which only differ in the bugfix number.
If `AllowUnsupported` is set also images with higher version numbers than the hard-coded minimum are allowed.